FirebasePhoneAuthUI/CountryCode/FUICountryCodes.m

```diff
@@ -163,19 +154,58 @@ - (FUICountryCodeInfo *)countryCodeInfoForPhoneNumber:(NSString *)phoneNumber {
   return nil;
 }
 
-- (FUICountryCodes *)searchCountriesByName:(NSString *)countryName {
-  if ([countryName length] == 0) {
-    return nil;
-  } else {
-    NSArray *results =
-        [self.countryCodesArray filteredArrayUsingPredicate:
-            [NSPredicate predicateWithFormat:FUI_JSON_COUNTRY_NAME_PREDICATE, countryName]];
-    return [[FUICountryCodes alloc] initWithCountriesArray:results];
+- (void)blackListCountries:(NSSet<NSString *> *)countries {
+  NSMutableArray<NSDictionary *> *array =
+      [[NSMutableArray alloc] initWithCapacity:_countryCodesArray.count];
+  for (NSDictionary *dict in self.countryCodesArray) {
+    NSString *countryCode = dict[FUI_JSON_COUNTRY_CODE_KEY];
+    NSString *dialCode = dict[FUI_JSON_DIALCODE_KEY];
+    if ([countries containsObject:countryCode] || [countries containsObject:dialCode]) {
+      continue;
+    }
+    [array addObject:dict];
   }
+  _countryCodesArray = array.mutableCopy;
+}
+
+- (void)whiteListCountries:(NSSet<NSString *> *)countries {
+  NSMutableArray *array = [[NSMutableArray alloc] initWithCapacity:_countryCodesArray.count];
+  for (NSDictionary *dict in self.countryCodesArray) {
+    NSString *countryCode = dict[FUI_JSON_COUNTRY_CODE_KEY];
+    NSString *dialCode = dict[FUI_JSON_DIALCODE_KEY];
+    if ([countries containsObject:countryCode] || [countries containsObject:dialCode]) {
+      [array addObject:dict];
+    }
+  }
+  _countryCodesArray = array.mutableCopy;
+}
+
+- (instancetype)searchCountriesByName:(NSString *)nameQuery {
+  NSArray<NSDictionary *> *results =
+      [self.countryCodesArray filteredArrayUsingPredicate:
+          [NSPredicate predicateWithFormat:FUI_JSON_COUNTRY_NAME_PREDICATE, nameQuery]];
+  return [[FUICountryCodes alloc] initWithCountriesArray:results];
 }
 
 #pragma mark Helper methods
 
+- (nullable instancetype)initWithCountriesArray:(NSArray<NSDictionary *> *)countries {
+  NSParameterAssert(countries);
+
+  if (!countries) {
+    return nil;
+  }
```

Comment:
This check should always succeed. Maybe it should check for `countries.count > 0` instead?

Reply:
Done.